### DISCUSSION TITLE
Upcoming features in 1.0 - 1.0 版本即将要完成的功能

### DISCUSSION THREAD

**Discussion by sanyuan0704** [sanyuan0704/island.js#73] [General]
感谢大家对 Island.js 的关注，目前核心的框架功能已经基本搭建完成([官网地址](https://island.sanyuan0704.top/))，感谢  @liuqh0609 @lweijian @shuaiqidekaige @STDSuperman 等等社区小伙伴的共建和支持。接下来还剩下一些功能需要开发，我们将在 1.0 的版本中完善以下的功能，并让 Island.js 的 API 和功能逐步稳定下来：

Markdown 编译部分:
- [x] 检测死链(dead link)，并在构建过程中给出提示，避免上线后链接打不开的问题https://github.com/sanyuan0704/island.js/issues/104
- [x] 支持 Custom container，包括 `:::tip`、`:::info`、`:::danger`和`:::warning` - 已分配至 @liuqh0609  https://github.com/sanyuan0704/island.js/pull/81
- [x] 代码块支持行号显示、指定部分行高亮 - 分配至 @Flower-F @gumingWu  https://github.com/sanyuan0704/island.js/pull/82
- [x] 支持 Emoji, 如`:tada:` -> 🎉，可参考 https://github.com/markdown-it/markdown-it-emoji/blob/master/lib/data/full.json @Flower-F  https://github.com/sanyuan0704/island.js/pull/101
- [x] 支持自定义 `target: _blank` 白名单，https://github.com/sanyuan0704/island.js/issues/102

CLI 部分:

- [x] 自定义 CLI 传参，透传至 Vite #109 
- [ ] unocss 配置文件移动至主题目录中，而不再放到 src/node，由主题开发者来定制
- [x] 支持自定义全局样式 https://github.com/sanyuan0704/island.js/issues/85
- [x] 支持插件机制 https://github.com/sanyuan0704/island.js/discussions/89 https://github.com/sanyuan0704/island.js/pull/122

运行时部分:

- [ ] 添加 preload 和 prefetch 优化 https://github.com/sanyuan0704/island.js/discussions/72
- [ ] 支持 hydrate 阶段 Island 组件懒加载 #111 

工作流部分:

- [ ] 依赖预构建，从而将众多依赖从 dependencies 中移除，优化安装体验
- [ ] Vitest 单测覆盖率完善，目前覆盖还比较少

文档部分:

- [ ] 编写 Island.js 与其它方案 Astro、Gatsby、VuePress/VitePress 的对比
- [x] 编写从零到一搭建文档站的教程（交互式教程，单独站点 https://github.com/sanyuan0704/island-tutorial）
    - [x] 初始化
    - [x] 导航栏、侧边栏的创建
    - [x] 书写 Markdown/MDX 文件
    - [x] 项目构建打包
    - [x] 接入国际化
    - [x] 项目部署上线
- [x] 编写自定义主题的开发教程
- [x] Client 部分的 API 文档，如 usePageData

默认主题部分:

- [x] Nav 导航栏的移动端版本
- [x] Sidebar 的移动端版本
- [x] 回到顶部组件开发 @gumingWu #98 
- [x] 图片点击放大组件开发 @liuqh0609  #106 
- [x] 支持路径 base 前缀，目前还不能和 i18n 一起使用
- [x] 搜索框优化 https://github.com/sanyuan0704/island.js/issues/90

如果感兴趣你可以在评论区认领功能，也可以尽情发表你的看法~

English Version:

Thank you for your attention to Island.js. At present, the core framework functions have been basically built ([official website address](https://island.sanyuan0704.top/)), thank you @liuqh0609 @lweijian @shuaiqidekaige @STDSuperman and other community small Cooperation and support of partners. There are still some functions left to develop. We will improve the following functions in version 1.0, and gradually stabilize the API and functions of Island.js:

Markdown compilation part:
- [x] Detect dead links and give prompts during the build process to avoid the problem that the link cannot be opened after going online https://github.com/sanyuan0704/island.js/issues/104
- [x] Support for Custom containers, including `:::tip`, `:::info`, `:::danger` and `:::warning` - assigned to @liuqh0609 https://github.com/ sanyuan0704/island.js/pull/81
- [x] Code block supports line number display and specified part of line highlighting - Assigned to @Flower-F @gumingWu https://github.com/sanyuan0704/island.js/pull/82
- [x] Support Emoji, such as `:tada:` -> 🎉, please refer to https://github.com/markdown-it/markdown-it-emoji/blob/master/lib/data/full.json
- [x] Support custom target: _blank config for link (https://github.com/sanyuan0704/island.js/issues/102)
- [ ] Support the use of global components in md(x), such as `NoSSR` components, without manually importing
```tsx
// mdx
<NoSSR>
  <MyComponent />
</NoSSR>
````

CLI section:

- [x] Custom CLI parameters, transparently transmitted to Vite #109 
- [ ] unocss configuration file moved to theme directory instead of src/node, customized by theme developer
- [x] Support custom global style https://github.com/sanyuan0704/island.js/issues/85
- [x] Support for plug-in mechanism https://github.com/sanyuan0704/island.js/discussions/89 https://github.com/sanyuan0704/island.js/pull/122

Runtime part:

- [ ] Add preload and prefetch optimization https://github.com/sanyuan0704/island.js/discussions/72
- [ ] The hydrate Island component can be loaded lazily #111 

Workflow section:

- [ ] Depends on pre-build, which removes many dependencies from dependencies and optimizes the installation experience
- [ ] Vitest single test coverage is perfect, but the coverage is still relatively small at present

Documentation section:

- [ ] Comparison of writing Island.js with other solutions Astro, Gatsby, VuePress/VitePress
- [x] Write a tutorial for building a documentation site from zero to one (interactive tutorial, separate site https://github.com/sanyuan0704/island-tutorial)
    - [x] Initialize
    - [x] Navigation bar, sidebar creation
    - [x] Write Markdown/MDX files
    - [x] Project build package
    - [x] Access Internationalization
    - [x] Project deployment goes live
- [x] Development tutorial for writing custom themes
- API documentation for [ ] Client section like usePageData

Default theme section:

- [x] Mobile version of Nav navigation bar @spicylemonhaha 
- [x] Mobile version of Sidebar @spicylemonhaha 
- [x] Back to Top Component Development @gumingWu #98 
- [x] Image click to enlarge component development @liuqh0609  #106  
- [x] supports path base prefix, currently not working with i18n
- [x] Search box optimization https://github.com/sanyuan0704/island.js/issues/90

If you are interested, you can claim the function in the comment area, or feel free to express your opinions~

**Comment by Flower-F** [comment]
@sanyuan0704 I would like to try the `代码块支持行号显示、指定部分行高亮`

**Reply by gumingWu** [reply]
sry about that I'm doing this task, but not commenting >_<

**Reply by Flower-F** [reply]
> sry about that I'm doing this task, but not commenting >_<

oh 不好意思刚才没看到，你弄好了哪些部分了，我指定部分行高亮的部分写完了。不如我先提个 pr，然后你来负责代码块支持行号显示的部分。

**Reply by callqh** [reply]
问题不大，感觉有时间也可以试试其他任务~🥷

**Reply by gumingWu** [reply]
> > sry about that I'm doing this task, but not commenting >_<
> 
> oh 不好意思刚才没看到，你弄好了哪些部分了，我指定部分行高亮的部分写完了。不如我先提个 pr，然后你来负责代码块支持行号显示的部分。

行号显示我也弄好了，准备会提pr

**Comment by callqh** [comment]
Maybe next I can try the dead link detection part

**Reply by sanyuan0704** [reply]
ok

**Comment by gumingWu** [comment]
`回到顶部组件开发`  get

**Comment by Flower-F** [comment]
Emoji support get
